### PR TITLE
PLANNER-2341: Add benchmarking example to Quarkus and Spring Boot School Timetabling

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Depends on https://github.com/kiegroup/optaplanner/pull/1249

### Referenced pull requests

* https://github.com/kiegroup/optaplanner/pull/1249

### CODE REVIEW

**Comment by triceo** [comment]
Any reason why this is still a draft?

**Review by triceo — APPROVED** [review]
Good to go once CI is green. (Which it currently isn't.)

**Comment by Christopher-Chianelli** [comment]
Probably be green tomorrow-ish; it doesn't look like it got the recent changes pushed to master

**Comment by Christopher-Chianelli** [comment]
Build chain can't find the `optaplanner.benchmark.solver.termination.spent-limit` property; rest of the actions passed.

**Comment by triceo** [comment]
It's not a random failure, it needs to be addressed.

**Comment by Christopher-Chianelli** [comment]
@triceo I can't reproduce it locally (I did the exact same maven command). Can you reproduce it?

File: spring-boot-school-timetabling/src/test/java/com/example/schooltimetabling/solver/TimeTableBenchmarkTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.example.schooltimetabling.rest.TimeTableController;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.optaplanner.benchmark.api.PlannerBenchmarkFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(properties = {
+        "optaplanner.benchmark.solver.termination.spent-limit=10s",
```
Comment (ge0ffrey) [inline spring-boot-school-timetabling/src/test/java/com/example/schooltimetabling/solver/TimeTableBenchmarkTest.java]: spent-limit between Quarkus and spring should be the same

File: quarkus-school-timetabling/src/test/java/org/acme/schooltimetabling/solver/TimeTableBenchmarkTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.acme.schooltimetabling.rest.TimeTableResource;
+import org.junit.jupiter.api.Test;
+import org.optaplanner.benchmark.api.PlannerBenchmarkFactory;
+
+import javax.inject.Inject;
+
+@QuarkusTest
+public class TimeTableBenchmarkTest {
```
Comment (ge0ffrey) [inline quarkus-school-timetabling/src/test/java/org/acme/schooltimetabling/solver/TimeTableBenchmarkTest.java:11]: Should these be a test? or a main app in src/test?

**Comment by Christopher-Chianelli** [comment]
...I think I figured out the problem. Looking at the logs, it using my old, outdated branch instead of master

**Comment by Christopher-Chianelli** [comment]
@triceo @ge0ffrey Fixed the issue. The issue was my branch on OptaPlanner was not updated after it was merged into master, which caused quickstarts to use an outdated branch (and thus not able to find the property)

**Pull request merged**